lib/Module/CMakeLists.txt

```diff
@@ -23,9 +23,14 @@ set(LLVM_COMPONENTS
   bitreader
   bitwriter
   codegen
+  instcombine
   ipo
   linker
+  mc
```

Comment:
Hmm, you are right. I blindly added everything what linker screamed is missing. So non-`llvm-config` configuration is more broken than `llvm-config`?

Really,  `llvm_map_components_to_libnames` returns less what it really should (opposing to llvm-config), it seems. No transitive closure of internal LLVM dependencies there. I'am at LLVM 4.